src/ims/v1p3/CatSession.php

```diff
@@ -89,11 +89,12 @@ public function __construct($engine, $sectionId, $data, SessionContext $context)
     public function getTestMap($results = [])
     {
         if (!empty($results) && $this->sessionState !== null) {
-            $context = [];
+            $itemResult = [array_pop($results)];
```

Comment:
Imho, `$itemResult = [end($results)];` would be better, because it isn't modify `$results` for no reason.

Reply:
optional: shouldn't we add unit tests for this ?

Reply:
and it is faster. agree

Reply:
What about the impact on the `::filterItemResults` method?